core/include/detray/masks/masks.hpp

```diff
@@ -181,6 +181,10 @@ class mask {
     DETRAY_HOST_DEVICE
     auto values() const -> const mask_values& { return _values; }
 
+    /// @returns the boundary values
+    DETRAY_HOST_DEVICE
+    auto values() -> mask_values& { return _values; }
```

Comment:
So the mask value can be changed after construction?

Reply:
Done, the masks are constructed directly now